tools/src/bin/rust-indexer.rs

```diff
@@ -809,3 +824,219 @@ fn main() {
+    let mut file = BufReader::new(file);
+    let mut file = protobuf::CodedInputStream::from_buf_read(&mut file);
+    let index = Index::parse_from(&mut file).expect("Failed to read scip index");
+
+    for doc in &index.documents {
+        let searchfox_path = Path::new(&doc.relative_path).to_owned();
+        let searchfox_path = match scip_prefix.and_then(|prefix| scip_file.strip_prefix(prefix).ok()) {
+            Some(p) => {
+                let mut p = p.to_owned();
+                p.pop();
+                p.join(&searchfox_path)
+            },
+            None => searchfox_path,
+        };
+
+        // let local_source_path = searchfox_path_to_local_path(searchfox_path, tree_info);
+        let output_file = tree_info.out_analysis_dir.join(&searchfox_path);
+        if let Err(err) = create_output_dir(&output_file) {
+            error!(
+                "Couldn't create dir for: {}, {:?}",
+                output_file.display(),
+                err
+            );
+            continue;
+        }
+        let mut file = match File::create(&output_file) {
+            Ok(f) => f,
+            Err(err) => {
+                error!(
+                    "Couldn't open output file: {}, {:?}",
+                    output_file.display(),
+                    err
+                );
+                continue;
+            }
+        };
+        // Be chatty about the files we're outputting so that it's easier to follow
+        // the path of rust analysis generation.
+        info!(
+            "Writing analysis for '{}' to '{}'",
+            searchfox_path.display(),
+            output_file.display()
+        );
+
+        // A map from local symbol to the index in doc.symbols for this document.
+        let mut doc_symbols_to_index = HashMap::new();
+        for (index, symbol) in doc.symbols.iter().enumerate() {
+            doc_symbols_to_index.insert(symbol.symbol.clone(), index);
+        }
+
+        let lookup_symbol = |s: &str| -> Cow<SymbolInformation> {
+            match doc_symbols_to_index.get(s) {
+                Some(i) => Cow::Borrowed(&doc.symbols[*i]),
+                None => {
+                    warn!("Didn't find symbol {:?} in local symbol table", s);
+                    // Fake it till you make it? We have no info for this
+                    // symbol, so...
+                    Cow::Owned(SymbolInformation {
+                        symbol: s.to_owned(),
+                        documentation: vec![],
+                        relationships: vec![],
+                        special_fields: Default::default(),
+                    })
+                }
+            }
+        };
+
+        for occurrence in &doc.occurrences {
+            let loc = scip_range_to_searchfox_location(&occurrence.range);
+            let symbol = lookup_symbol(&occurrence.symbol);
+            {
+                let global = sanitize_symbol(&symbol.symbol);
+                let pretty = pretty_symbol(&symbol.symbol);
```

Comment:
For [source symbols](https://github.com/mozsearch/mozsearch/blob/master/docs/analysis.md#sources) we expect pretty to have a space-delimited human-readable description before the actual pretty symbol name, for example "constructor SYMBOL".

The [existing code does this](https://github.com/mozsearch/mozsearch/blob/b7c0f39354fe3ec74a74d3ad4937147ab29c213c/tools/src/bin/rust-indexer.rs#L325-L341) so it would probably be a regression.  Looking at the rust analysis check outputs ([1](https://github.com/mozsearch/mozsearch/blob/master/tests/tests/checks/snapshots/analysis/rust/simple.rs/check_glob%40simple_Loader_new__json.snap) [2](https://github.com/mozsearch/mozsearch/blob/master/tests/tests/checks/snapshots/analysis/rust/generated.rs/check_glob%40GeneratedType__json.snap) [3](https://github.com/mozsearch/mozsearch/blob/master/tests/tests/checks/snapshots/analysis/rust/test_rust_dependency/check_glob%40MyType__json.snap)) I only see target records.

It turns out the reason we don't have source records in there is that we don't populate the "syntax" field [which is used to determine the "kind" of source records by the `filter-analysis` command](https://github.com/mozsearch/mozsearch/blob/b7c0f39354fe3ec74a74d3ad4937147ab29c213c/tools/src/cmd_pipeline/cmd_filter_analysis.rs#L63-L83).

Reply:
Given this won't be used in m-c just yet I'll merge this for now and work on extra improvements for this and syntax in follow-ups.

Reply:
Makes sense!  Feel free to land any follow-ups without additional review unless you particularly are looking for feedback.